### ISSUE TITLE
Improve vm.js wrapper to handle sandboxing properly

### ISSUE THREAD

**Issue by kaustavdm** [getgauge-contrib/gauge-js#43]
Based on the comments in #39, the `vm.js` script needs modifications to suit the needs of a proper sandbox.

**Comment by kaustavdm** [comment]
@renjithgr you up for this? If so, please self-assign.

**Comment by kaustavdm** [comment]
@renjithgr I have implemented a custom VM and it works for me. Tested on nodejs v4.4.1. Can you test on your end?

**Comment by renjithgr** [comment]
Great job! Works on nodejs v5.5.0! :+1:

**Comment by kaustavdm** [comment]
Need to figure out what is the minimum NodeJS version that we can support because of this change. According to Node's API docs, you will need v0.12.x+. But, we still need to confirm it.

**Comment by kaustavdm** [comment]
I'm still seeing trouble on Windows :angry:

**Comment by kaustavdm** [comment]
`CWD` in test implementations is changed to `GAUGE_PROJECT_ROOT`. Reading files from relative to the project root works on Linux and OSX, but fails on Windows.

Here is the code for `tests/step_implementations.js`:

```js
var fs = require("fs");

// I know you can `require()` this, but just for the sake of the example...
// This should read the `manifest.json` file at the root of the project,
// but it ends up looking in the parent directory of the project root in Windows.
var manifest = fs.readFileSync("manifest.json").toString("utf-8");
```

**Comment by kaustavdm** [comment]
So, it seems like I solved this as part of e87f93163fd7ed1d84488ba9a5a6e0c893ecb74e :wink:

**Comment by philipbeadle** [comment]
Am I doing this right?  I have this file https://github.com/philipbeadle/gauge-js-angular-webdriver-demo/blob/1-Using_module.exports_to_hold_Driver/tests/tests/driver.js
But I cant reference the driver variable in 
https://github.com/philipbeadle/gauge-js-angular-webdriver-demo/blob/1-Using_module.exports_to_hold_Driver/tests/tests/angular.js

The driver.js file doesn't appear to be executed at all as the console.log doesn't write out either.
